contracts/campaignAirdrops/ExistingERC20DropCampaign.sol

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.15;
+
+import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "./shared/CampaignAirdropsShared.sol";
+
+/// @title Airdrops existing ERC20 tokens for airdrop recipients
+contract ExistingERC20DropCampaign is CampaignAidropsShared {
+    using SafeERC20 for IERC20;
+
+    IERC20 public immutable rewardToken;
+    uint256 public immutable tokenSupply;
+
+    string public constant airdropType = "ERC20";
+    uint256 internal numberOfClaimers;
+    uint256 public tokensPerClaim;
+    uint256 public airdropExpirationTimestamp;
+
+    address internal airdropFunder;
+
+    event MerkleRootSet(bytes32 merkleRoot, uint256 _numberOfClaimers);
+
+    error AlreadyFunded();
+    error AirdropStillActive();
+    error AirdropExpired();
+
+    constructor(
+        address _rewardToken,
+        uint256 _tokenSupply,
+        address _airbroCampaignFactoryAddress
+    ) CampaignAidropsShared(_airbroCampaignFactoryAddress) {
+        rewardToken = IERC20(_rewardToken);
+        tokenSupply = _tokenSupply;
+    }
+
+    /// @notice Sets the merkleRoot and the number of claimers (also setting the amount each claimer receivers).
+    /// Can only be done by admin.
+    /// @dev Implements a handler method from the parent contract for performing checks and changing state
+    /// @param _merkleRoot The root hash of the Merle Tree
+    /// @param _numberOfClaimers The number of users eligible to claim
+    function setMerkleRoot(bytes32 _merkleRoot, uint256 _numberOfClaimers) external onlyAdmin {
+        super.setMerkleRootHandler(_merkleRoot);
+
+        airdropExpirationTimestamp = block.timestamp + 60 days;
+        numberOfClaimers = _numberOfClaimers;
+        tokensPerClaim = tokenSupply / _numberOfClaimers;
+
+        emit MerkleRootSet(_merkleRoot, _numberOfClaimers);
```

Comment:
This kinda belongs to parents `setMerkleRootHandler`. Here you are mixing apple and peaches.
Number of claimers has nothing to do with merkle root.
I'd emit `emit MerkleRootSet(_merkleRoot)` inside of `super.setMerkleRootHandler`, and remove it from child contracts.
For this particular use case maybe use different event? emit NumberOfClaimersSet or whatever

Reply:
I thought about this, and I think you are right. Will update.